src/bytes.rs

```diff
@@ -113,54 +131,75 @@ impl<'a> Bytes<'a> {
     /// Unlike [`Bytes::from_static`] and `<Bytes as From<&'static ..>>::from` implementations, the resulting [`Bytes`] cannot tell that it's referring to `'static` data even if it is, and will therefore perform a copy when [`Bytes::upgrade`]ing.
     /// ```
     /// # use safer_ffi::bytes::Bytes;
-    /// let data = "Hello there, this string is long enough that it'll cross the inline-threshold (core::mem::size_of::<Bytes>() - 1) on all supported platforms";
+    /// let data = "Hello there, this string is long enough that it'll cross the inline-threshold (mem::size_of::<Bytes>() - 1) on all supported platforms";
     /// let mut bytes = Bytes::from_slice(data.as_bytes());
     /// assert!(bytes.upgrade_will_allocate());
     /// let data = "This string isn't";
     /// let mut bytes = Bytes::from_slice(data.as_bytes());
     /// assert!(!bytes.upgrade_will_allocate());
     /// ```
     ///
-    /// Note that if the slice is small enough to be interned in the [`Bytes`], it'll be, allowing for free upgrades. You may use the [`Bytes::intern_slice`] constructor instead of this one if you want to be able to handle interning not being possible.
-    pub fn from_slice(data: &'a [u8]) -> Self {
-        static VT: BytesVt = BytesVt {
+    /// Note that if the slice is small enough to be inlined in the [`Bytes`], it'll be, allowing for free upgrades. You may use the [`Bytes::inline_slice`] constructor instead of this one if you want to be able to handle inlining not being possible.
+    pub const fn from_slice(data: &'a [u8]) -> Self {
+        const VT: BytesVt = BytesVt {
```

Comment:
Is the address of the resulting `self.vtable` ever compared against another? I imagine not, but just checking since "downgrading" a `static` to a `const` does break the uniqueness guarantee

Reply:
it's not, there wouldn't be a point in downcasting to a slice :)